src/main/kotlin/no/nav/tilleggsstonader/sak/vilkår/vilkårperiode/evaluering/EvalueringAktivitet.kt

```diff
@@ -0,0 +1,88 @@
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.DelvilkårAktivitet
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.DelvilkårVilkårperiode.Vurdering
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.ResultatDelvilkårperiode
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.ResultatVilkårperiode
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.SvarJaNei
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.dto.DelvilkårAktivitetDto
+
+object EvalueringAktivitet {
+    fun utledResultat(
+        type: AktivitetType,
+        delvilkår: DelvilkårAktivitetDto,
+    ): ResultatEvaluering {
+        val oppdatertDelvilkår = utledVurderingerDelvilkår(type, delvilkår)
+        val resultatAktivitet = resultatVilkårperiode(oppdatertDelvilkår)
+        return ResultatEvaluering(oppdatertDelvilkår, resultatAktivitet)
+    }
+
+    private fun utledVurderingerDelvilkår(
+        type: AktivitetType,
+        delvilkår: DelvilkårAktivitetDto,
+    ): DelvilkårAktivitet {
+        val vurderingLønnet = vurderingLønnet(type, delvilkår.lønnet)
+        val vurderingMottarSykepenger = vurderingMottarSykepenger(delvilkår.mottarSykepenger)
+
+        return DelvilkårAktivitet(
+            lønnet = vurderingLønnet,
+            mottarSykepenger = vurderingMottarSykepenger,
+        )
+    }
+
+    private fun resultatVilkårperiode(
+        delvilkår: DelvilkårAktivitet,
+    ): ResultatVilkårperiode {
+        val vurderingLønnet = delvilkår.lønnet
+        val vurderingMottarSykepenger = delvilkår.mottarSykepenger
+        val resultater = listOf(vurderingLønnet.resultat, vurderingMottarSykepenger.resultat)
+            .filterNot { it == ResultatDelvilkårperiode.IKKE_AKTUELT }
+
+        return when {
+            resultater.contains(ResultatDelvilkårperiode.IKKE_VURDERT) -> ResultatVilkårperiode.IKKE_VURDERT
+            resultater.contains(ResultatDelvilkårperiode.IKKE_OPPFYLT) -> ResultatVilkårperiode.IKKE_OPPFYLT
+            resultater.all { it == ResultatDelvilkårperiode.OPPFYLT } -> ResultatVilkårperiode.OPPFYLT
+            else -> error(
+                "Ugyldig resultat resultatLønnet=$vurderingLønnet" +
+                    " resultatMottarSykepenger=$vurderingMottarSykepenger",
+            )
+        }
+    }
+
+    private fun vurderingLønnet(type: AktivitetType, svar: SvarJaNei?): Vurdering {
+        return when (type) {
+            AktivitetType.TILTAK -> Vurdering(svar, utledResultatLønnet(svar))
+            AktivitetType.UTDANNING,
+            AktivitetType.REELL_ARBEIDSSØKER,
+            -> ikkeVurdertLønnet(type, svar)
+        }
+    }
+
+    private fun ikkeVurdertLønnet(type: AktivitetType, svar: SvarJaNei?): Vurdering {
+        feilHvis(svar != null) {
+            "Ugyldig svar=$svar for lønnet for $type"
+        }
+        return Vurdering(svar = svar, resultat = ResultatDelvilkårperiode.IKKE_AKTUELT)
```

Comment:
Burde man her sette svar til `IKKE_AKTUELT` i stedet for å sende svar videre? Svar burde være det, men kanskje noe skjer på veien?

Reply:
Så nå at det sjekkes at svar er `null`, skulle ikke den vært `IKKE_AKTUELT`?

Reply:
Dette ble litt mye for meg, `IKKE_AKTUELT` er et resultat og ikke et svar som jeg tenkte da jeg skrev disse hehe <img src="https://media0.giphy.com/media/1Y7ChRtbWnYONjDidg/giphy.gif"/>

Reply:
Diskutert denne, vi gikk for resultat = IKKE_AKTUELT